### PR TITLE
use go 1.13

### PR DESCRIPTION
**Pull request by techknowlogick**
as title

### CODE REVIEW

**Review by sapk — CHANGES_REQUESTED** [review]
You need to update go.mod by upgrading the first line to enable 1.13 libs.

**Comment by techknowlogick** [comment]
@sapk seems golangci-lint fails, and the reports contain incorrect information. I will look into this more.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module code.gitea.io/gitea
 
-go 1.12
+go 1.13
```
Comment (lafriks) [inline go.mod:3]: why update go.mod?

**Comment by lafriks** [comment]
> 
> 
> You need to update go.mod by upgrading the first line to enable 1.13 libs.

why?

**Comment by sapk** [comment]
@lafriks I need to find the link but I remember that the std lib are also versionned now and that go compiler will use the one specified in go.mod

**Comment by sapk** [comment]
From https://golang.org/doc/go1.13
> If your code uses modules and your go.mod files specifies a language version, be sure it is set to at least 1.13 to get access to these language changes. You can do this by editing the go.mod file directly, or you can run go mod edit -go=1.13.

But I have also seen that some suggest to declare the minimal go version supported. This seems logic since it should be the lowest required std libs. I don't know go internal to say what is the best. 

On a sidenote, I hope we can use new error methods soon.

**Comment by lafriks** [comment]
> 
> 
> From https://golang.org/doc/go1.13
> 
> > If your code uses modules and your go.mod files specifies a language version, be sure it is set to at least 1.13 to get access to these language changes. You can do this by editing the go.mod file directly, or you can run go mod edit -go=1.13.
> 
> But I have also seen that some suggest to declare the minimal go version supported. This seems logic since it should be the lowest required std libs. I don't know go internal to say what is the best.
> 
> On a sidenote, I hope we can use new error methods soon.

I should more carefully read release notes :D Thanks for pointing out to this ;)

**Comment by sapk** [comment]
@lafriks go module is a good point providing a standard and more secure deps management but they are few quirks to get around and I also don't get some.

**Comment by techknowlogick** [comment]
Currently blocked on: https://github.com/golangci/golangci-lint/issues/663

**Comment by guillep2k** [comment]
> 
> 
> Currently blocked on: [golangci/golangci-lint#663](https://github.com/golangci/golangci-lint/issues/663)

Just in case, that issue was marked as duplicate of [golangci/golangci-lint#659](https://github.com/golangci/golangci-lint/issues/659)

**Comment by techknowlogick** [comment]
deps have been upgraded and build now passes. In addition to golangci-lint, I also update swagger.

Please review :)

**Review by sapk — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**